docs/reference/cli.md

```diff
@@ -44,314 +47,433 @@ The line `Telemetry enabled` disappears when you disable it.
-`--overwrite` (boolean) Overwrites existing index.
 
-*Examples*
 
-*Creating a new index on local file system*
+# index
+Index operation commands such as `create`, `index`, or `search`...
 
-```bash
-quickwit new --index-uri file:///quickwit-indexes/catalog --index-config-path ~/quickwit-conf/index_config.json
-```
+### index create
 
-Creating a new index on Amazon S3*
+Creates an index at `index-uri` configured by a yaml file located at `index-config-uri`. The command fails if an index already exists at `index-uri` unless `overwrite` is passed. 
+When `overwrite` is enabled, the command deletes all the files stored at `index-uri` before creating a new index. 
+The index config defines how a document and fields it contains, are stored and indexed, see the [index config documentation](index-config.md).
+  
+`quickwit index create [args]`
 
-```bash
-quickwit new --index-uri s3://quickwit-indexes/catalog --index_config-path ~/quickwit-conf/index_config.json
-```
-
-*Replacing an existing index*
+*Synopsis*
 
 ```bash
-quickwit new --index-uri s3://quickwit-indexes/catalog --index_config-path ~/quickwit-conf/index_config.json --overwrite
+quickwit index create
+    --index <index>
+    --index-config <index-config>
+    [--index-uri <index-uri>]
+    --config <config>
+    [--overwrite]
 ```
 
-:::note
-
-When creating an index on a local file system, absolute path is enforce. This implies that index-uri like `file:///quickwit-indexes/catalog` pertenains you have the required permissions on `/quickwit-indexes/catalog`.
+*Options*
 
-:::
+`--index` ID of the target index.    
+`--index-config` Location of the index config file.    
+`--index-uri` Index data (or splits) storage URI.    
+`--config` Quickwit config file.    
+`--overwrite` Overwrites pre-existing index.    
 
-### Index
+*Examples*
 
-*Description*
+*Create a new index with metastore in the local `metastore` folder.*
+```bash
+quickwit index create --index-config-uri ./wikipedia_index_config.yaml  --config=quickwit.yaml
+```
 
-Indexes a dataset consisting of newline-delimited JSON objects located at `input-path` or read from *stdin*. The data is appended to the target index specified by `index-uri` unless `overwrite` is passed. `input-path` can be a file or another command output piped into stdin. Currently, only local datasets are supported. By default, tantivy's indexer will work with a heap of 1 GiB of memory, but this can be set with the `heap-size` options. This does not directly reflect the overall memory usage of `quickwit index`, but doubling this value should give a fair approximation.
+### index ingest
 
+Indexes a dataset consisting of newline-delimited JSON objects located at `input-path` or read from *stdin*. 
+The data is appended to the target index specified by `index-uri` unless `overwrite` is passed. `input-path` can be a file or another command output piped into stdin. 
+Currently, only local datasets are supported. 
```

Comment:
`Currently, only local datasets are supported.`. I think this has changed, it can work with a file sitting on S3.

Reply:
internally this is a PathBuf, it also uses `exists()`

```
    pub data_dir_path: PathBuf,
```

Reply:
I was not talking about `data_dir_path` which is different.
Anyway, I just checked that and the answer is `FileSource` only works with a local file.